cpu/cortexm_common/include/systick.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2020 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_periph_systick SysTick Timer
+ * @ingroup     cpu_cortexm_common
+ * @ingroup     drivers_periph
+ * @brief       Low-level timer based on SysTick
+ *
+ * SysTick is a 24 bit down-counting timer that is implemented on every Cortex-M
+ * processor. It runs at the same frequency as the CPU and will generate an
+ * interrupt when it reaches zero.
+ *
+ * To comply with the RIOT timer interface that expects an up-counting, monotonic
+ * timer, SysTick is augmented by software to fulfill these requirements.
+ * This also adds a software prescaler to simulate arbitrary frequencies and introduces
```

Comment:
Is this necessary and / or desirable?

We already have __two__ higher level timer implementations that do exactly that, and more.

There are quite some challenges to overcome to get this right.

As is, I'd probably cut out the guts and re-implement as ztimer backend.

Reply:
The issue was that xtimer cannot handle arbitrary frequencies: https://github.com/RIOT-OS/RIOT/pull/14553#discussion_r457185983

I'm personally fine with having this a `ztimer` only. IMO, `xtimer` is on the way out anyway.

Reply:
I found it useful to just being able to use `xtimer_on_systick` to free up the periph timer and not having to configure ztimer as well.
I don't think the conditional overflow handling adds much to the code, and it's pretty minimal when enabled.

Reply:
ztimer still expects the timer to be up-counting, right?